src/bournemouth/resources.py

```
@@ -230,6 +169,76 @@ async def handle(request: ChatWsRequest) -> None:
+
+    async def _get_api_key(self) -> str | None:
+        if self._user is None:
+            raise RuntimeError("User must be set before calling _get_api_key")
+        try:
+            _, api_key = await load_user_and_api_key(self.session_factory, self._user)
+        except falcon.HTTPUnauthorized:
+            return None
+        return api_key
+
+
+    @handles_message("chat")
```

Flake8 (7.2.0)

[error] too many blank lines (2)

(E303)